physics/GWD/drag_suite.F90

```diff
@@ -1414,6 +1417,1228 @@ subroutine drag_suite_run(                                           &
+                     kblk(i)= k
+                     zblk = zl(i,k)
+                     RDXZB(i) = real(k,kind=kind_phys)
+                     flag(i) = .false.
+                  endif
+               endif
+            enddo
+            if(.not.flag(i)) then
+!
+!--------- compute flow-blocking stress
+!
+               cd = max(2.0-1.0/od(i),cdmin)
+               taufb(i,kts) =  0.5 * roll(i) * coefm(i) /                  &
+                                 max(dxmax_ls,dxy(i))**2 * cd * dxyp(i) *  &
+                                 olp(i) * zblk * ulow(i)**2
+               tautem = taufb(i,kts)/float(kblk(i)-kts)
+               do k = kts+1, kpblmax
+                  if (k .le. kblk(i)) taufb(i,k) = taufb(i,k-1) - tautem
+               enddo
+!
+!   reset gwd stress below blocking layer
+!
+               do k = kts,kpblmax
+                if (k .le. kblk(i)) taup(i,k) = taup(i,kblk(i))
+               enddo
+!              if(kblk(i).gt.5) print *,' gwd kbl komax kbmax kblk ',kbl(i),komax(i),kbmax(i),kblk(i)
+!              if(kblk(i).gt.5) print *,' gwd elvmax zlowtop zblk ',elvmax(i),zlowtop(i),zl(i,kblk(i))
+            endif
+
+         endif   ! if (.not.ldrag(i))
+
+      endif   ! if ( ls_taper(i).GT.1.E-02 )
+
+   enddo   ! do i=its,im
+
+ENDIF   ! IF ( (do_gsl_drag_ls_bl) .and. (gwd_opt_bl .EQ. 1) )
+!===========================================================
+IF ( (do_gsl_drag_ls_bl) .and.                                       &
+     (gwd_opt_ls .EQ. 1 .OR. gwd_opt_bl .EQ. 1) ) THEN 
+
+   do i=its,im
+
+      if ( ls_taper(i) .GT. 1.E-02 ) then
+
+!
+!  calculate - (g)*d(tau)/d(pressure) and deceleration terms dtaux, dtauy
+!
+         do k = kts,km
+            taud_ls(i,k) = 1. * (taup(i,k+1) - taup(i,k)) * csg / del(i,k)
+            taud_bl(i,k) = 1. * (taufb(i,k+1) - taufb(i,k)) * csg / del(i,k)
+         enddo
+!
+!  limit de-acceleration (momentum deposition ) at top to 1/2 value
+!  the idea is some stuff must go out the 'top'
+         do klcap = lcap,km
+            taud_ls(i,klcap) = taud_ls(i,klcap) * factop
+            taud_bl(i,klcap) = taud_bl(i,klcap) * factop
+         enddo
+!
+!  if the gravity wave drag would force a critical line
+!  in the lower ksmm1 layers during the next deltim timestep,
+!  then only apply drag until that critical line is reached.
+!
+         do k = kts,kpblmax-1
+           if ((taud_ls(i,k)+taud_bl(i,k)).ne.0.)                  then
+               dtfac(i,k) = min(dtfac(i,k),abs(velco(i,k)                 &
+                       /(deltim*rcs*(taud_ls(i,k)+taud_bl(i,k)))))
+           endif
+         enddo
+! apply limiter to mesosphere drag, reduce the drag by density factor 10-3
+! prevent wind reversal...
+!
+         do k = kpblmax,km
+           if ((taud_ls(i,k)+taud_bl(i,k)).ne.0..and.prsl(i,k).le.pcutoff) then
+               denfac = min(ro(i,k)/pcutoff_den,1.)
+               dtfac(i,k) = min(dtfac(i,k),denfac*abs(velco(i,k)          &
```

Comment:
@JongilHan66 This is the line causing the latest trouble in https://github.com/ufs-community/ufs-weather-model/pull/2290. The problem is that this look is going up to 'km' when 'velco' is only defined up to 'km-1'. Would it work to simply change the loop from `kpblmax' to 'km-1', or does this mess up something else?

Reply:
> Let me know, and I can make the change and test that it works real quick.

@grantfirl I don't think the change causes a problem. Please change the loop from kpblmax to km-1.
BTW, in 'drag_suite.meta', 'vtype' and 'psl_gwd_dx_factor' were removed from the original meta file. Is it ok?

Reply:
@JongilHan66 Yes, they should have been removed because `drag_suite_psl()` is not actually exposed to the CCPP. It is only being called from within `ugwpv1_gsldrag_run` or `unified_ugwp_run`. Both the ugwpv1_gsldrag and unified_ugwp schemes have those two variables accessible to pass in to drag_suite_psl already.

Reply:
> @JongilHan66 Yes, they should have been removed because `drag_suite_psl()` is not actually exposed to the CCPP. It is only being called from within `ugwpv1_gsldrag_run` or `unified_ugwp_run`. Both the ugwpv1_gsldrag and unified_ugwp schemes have those two variables accessible to pass in to drag_suite_psl already.

@grantfirl I see. Thanks!!